libraries/rush-lib/src/logic/operations/ShellOperationRunner.ts

```diff
@@ -450,6 +451,23 @@ export class ShellOperationRunner implements IOperationRunner {
                   additionalContext['$' + varName] = process.env[varName] || '';
                 }
               }
+              if (operationSettings.dependsOnAdditionalFiles) {
+                const additionalFiles: Map<string, string> =
+                  await AdditionalFilesHasher.getFileHashedFilesMap(
+                    operationSettings.dependsOnAdditionalFiles,
+                    this._rushProject.projectFolder
+                  );
+
+                terminal.writeDebugLine(
+                  `Including additional files to calculate build cache hash:\n  ${Array.from(
+                    additionalFiles.keys()
+                  ).join('\n  ')} `
+                );
+
+                for (const [filePath, fileHash] of additionalFiles) {
+                  additionalContext['file://' + filePath] = fileHash;
```

Comment:
Here `'file://'` is just an arbitrary prefix, just to make sure that files will not clash with env variables.